jekyll/_cci2/controlling-serial-execution-across-your-organization.adoc

```diff
@@ -0,0 +1,227 @@
+
+. Add the `serial-group` key to each job across your organization that should run in sequence, using the same value for all related jobs:
++
+[,yml]
+----
+include::../_includes/snippets/orchestration-examples/serial-group.yml[]
+----
+
+. Run your pipelines and observe the sequential execution on the pipelines page in the CircleCI web app.
+
+== How serial groups work
+
+When you configure a job with a `serial-group` key, CircleCI's platform treats the job differently from non-sequential group jobs. The following sections outline what happens behind the scenes.
+
+=== Job identification and queuing
+
+CircleCI identifies jobs that belong to the same serial group by the string value you provide. When multiple jobs across your organization use identical serial group values, they enter a shared queue. This queue operates on a first-in, first-out basis but with special priority rules.
+
+=== Lock Mechanism
+
+CircleCI uses a distributed locking system to control execution order. When a job with a serial group starts, it attempts to acquire a "lock" for that specific serial group value. If no other job holds the lock, the job proceeds. If another job already holds the lock, subsequent jobs must wait in the queue.
+
+=== Pipeline number priority
+
+CircleCI uses pipeline numbers to determine execution priority. Jobs from newer pipelines (higher pipeline numbers) take precedence over jobs from older pipelines within the same project. If a job from an older pipeline tries to run while a newer pipeline's job is waiting or running, the older job is automatically skipped.
+
+=== Job state transitions
+
+Jobs in a serial group move through several states:
+
+* First, the job enters a "queued" state.
+* Once a job starts (reaches its place in the queue) it transitions to a "running" state.
+* Upon completion (whether successfully, with failure, or by cancellation) the next job in the queue starts. The queue continues to process jobs even if some jobs fail.
+
+=== Time limits
+
+To prevent indefinite waits, jobs can remain in the queue for up to five hours. After this time, CircleCI automatically cancels the waiting job and removes it from the queue.
+
+This architecture provides a robust system for coordinating work across your organization while maintaining execution order integrity and preventing deployment conflicts or race conditions in shared environments.
+
+== Serial Groups in the CircleCI UI
+
+When using serial groups in your CircleCI configuration, sequential group start and end blocks are visible when viewing your workflow graphs.
+
+=== Identifying serial groups
+
+Serial group jobs are displayed in the CircleCI web app in workflow graphs with the following visual elements:
+
+* A job in a serial group is preceded by a block with a lock icon to show the start of the serial group. A serial group end block follows the job.
```

Comment:
~~ Should this be key icon? Similar mentions below, but I won't comment on all of them.

```suggestion
* A job in a serial group is preceded by a block with a key icon to show the start of the serial group. A serial group end block follows the job.
```

Reply:
Oh yes good spot! Thanks